src/visitors/semantic/completion/mod.rs

```diff
@@ -7,14 +7,91 @@ use crate::protocol::responses::{
     CompletionItem, CompletionItemKind, InsertTextFormat,
 };
 use crate::shared::signatures::get_argument_names;
-use crate::shared::RequestContext;
+use crate::shared::{Function, RequestContext};
 use crate::stdlib::{create_function_signature, Completable};
 use crate::visitors::semantic::utils;
 
 use flux::semantic::nodes::*;
 use flux::semantic::types::MonoType;
 use flux::semantic::walk::{Node, Visitor};
 
+pub struct FunctionFinderState {
+    pub functions: Vec<Function>,
+}
+
+pub struct FunctionFinderVisitor {
+    pub pos: Position,
+    pub state: Arc<Mutex<FunctionFinderState>>,
+}
+
+impl FunctionFinderVisitor {
+    pub fn new(pos: Position) -> Self {
+        FunctionFinderVisitor {
+            pos,
+            state: Arc::new(Mutex::new(FunctionFinderState {
+                functions: vec![],
+            })),
+        }
+    }
+}
+
+impl<'a> Visitor<'a> for FunctionFinderVisitor {
+    fn visit(&mut self, node: Rc<Node<'a>>) -> bool {
+        if let Ok(mut state) = self.state.lock() {
+            let loc = node.loc();
+            let pos = self.pos.clone();
+
+            if loc.start.line > pos.line + 1
```

Comment:
wdyt about extending position again with a method that encapsulated this position comparison logic that we see throughout the file?

Reply:
good call

Reply:
I decided to define a function with this logic rather than giving position any knowledge of a flux SourceLocation